На Windows не сохраняется кэш MapDB: VolumeIOError при сохранении HashStorage

### Проблема

На Windows при сохранении кэша `HashStorage` после сборки/запуска тестов раннер падает на `MapDB commit()`.

Сама сборка и тесты при этом могут завершаться успешно, но состояние кэша не сохраняется, и на следующем запуске кэш перестраивается заново.

### Как проявляется

В логе после успешной сборки появляется предупреждение:

- `Не удалось сохранить хеши файлов (...) - кэш будет перестроен при следующей сборке`
- `Не удалось сохранить временную метку для 'edt' - кэш будет перестроен при следующей сборке`

Далее в стеке:

- `org.mapdb.DBException$VolumeIOError`
- `org.mapdb.volume.MappedFileVol.truncate`
- `org.mapdb.WriteAheadLog.destroyWalFiles`
- `Caused by: java.io.IOException: Запрошенную операцию нельзя выполнить для файла с открытой пользователем сопоставленной секцией`

### Ожидаемое поведение

Кэш хешей и временные метки должны успешно сохраняться после сборки/прогона тестов, без ошибок `VolumeIOError` и без перестроения кэша на следующем запуске.

### Фактическое поведение

На Windows MapDB не может завершить `commit()` и удалить/усечь WAL-файл. Похоже на проблему `memory-mapped files + truncate/delete` под Windows.

### Среда

- OS: Windows
- Java: 17.0.13
- runner: `0.5.1`
- проект: EDT
- сценарий: `build_project` / `run_all_tests`

### Шаги воспроизведения

1. Запустить test-runner на Windows.
2. Выполнить `run_all_tests` или `build_project` для EDT-проекта.
3. Дождаться завершения сборки.
4. Посмотреть лог сохранения состояния подпроекта.

### Фрагмент лога

```text
15:55:36.528 DEBUG - Сохранение состояния подпроекта: LicenseManager
15:55:36.651 WARN  - Не удалось сохранить хеши файлов (4644 шт.) - кэш будет перестроен при следующей сборке
15:55:36.652 DEBUG - Детали ошибки сохранения хешей
org.mapdb.DBException$VolumeIOError: null
    at org.mapdb.volume.MappedFileVol.truncate(MappedFileVol.java:285)
    at org.mapdb.WriteAheadLog.destroyWalFiles(WriteAheadLog.java:791)
    at org.mapdb.StoreWAL.commit(StoreWAL.kt:645)
    at org.mapdb.DB.commit(DB.kt:435)
    at io.github.alkoleft.mcp.infrastructure.storage.HashStorage.batchUpdate(HashStorage.kt:112)
    at io.github.alkoleft.mcp.infrastructure.storage.FileBuildStateManager.updateHashes(FileBuildStateManager.kt:89)
    at io.github.alkoleft.mcp.application.actions.change.FileSystemChangeAnalysisAction.saveSourceSetState(FileSystemChangeAnalysisAction.kt:145)
Caused by: java.io.IOException: Запрошенную операцию нельзя выполнить для файла с открытой пользователем сопоставленной секцией
    at java.base/sun.nio.ch.FileDispatcherImpl.truncate0(Native Method)
    at java.base/sun.nio.ch.FileDispatcherImpl.truncate(FileDispatcherImpl.java:90)
    at java.base/sun.nio.ch.FileChannelImpl.truncate(FileChannelImpl.java:437)
    at org.mapdb.volume.MappedFileVol.truncate(MappedFileVol.java:283)

15:55:36.683 WARN  - Не удалось сохранить временную метку для 'edt' - кэш будет перестроен при следующей сборке
15:55:36.683 DEBUG - Детали ошибки сохранения временной метки
org.mapdb.DBException$VolumeIOError: null
    at org.mapdb.volume.MappedFileVol.truncate(MappedFileVol.java:285)
    at org.mapdb.WriteAheadLog.destroyWalFiles(WriteAheadLog.java:791)
    at org.mapdb.StoreWAL.commit(StoreWAL.kt:645)
    at org.mapdb.DB.commit(DB.kt:435)
    at io.github.alkoleft.mcp.infrastructure.storage.HashStorage.storeTimestamp(HashStorage.kt:137)
Caused by: java.io.IOException: Запрошенную операцию нельзя выполнить для файла с открытой пользователем сопоставленной секцией
    at java.base/sun.nio.ch.FileDispatcherImpl.truncate0(Native Method)
    at java.base/sun.nio.ch.FileDispatcherImpl.truncate(FileDispatcherImpl.java:90)
    at java.base/sun.nio.ch.FileChannelImpl.truncate(FileChannelImpl.java:437)
    at org.mapdb.volume.MappedFileVol.truncate(MappedFileVol.java:283)

15:55:36.710 WARN  - Could not delete file: C:\Users\aleksandr.ponkratov\AppData\Local\Temp\codex-yaxunit-actual-20260312_155355\mcp-yaxunit-runner\edt-project\hash-storages\designer.db.wal.0
15:55:36.717 WARN  - Could not delete file: C:\Users\aleksandr.ponkratov\AppData\Local\Temp\codex-yaxunit-actual-20260312_155355\mcp-yaxunit-runner\edt-project\hash-storages\designer.db.wal.0

--- a/src/main/kotlin/io/github/alkoleft/mcp/infrastructure/storage/HashStorage.kt
+++ b/src/main/kotlin/io/github/alkoleft/mcp/infrastructure/storage/HashStorage.kt
@@ -21,6 +21,8 @@
 
 package io.github.alkoleft.mcp.infrastructure.storage
 
+import io.github.alkoleft.mcp.application.core.PlatformType
+import io.github.alkoleft.mcp.infrastructure.utility.PlatformDetector
 import io.github.oshai.kotlinlogging.KotlinLogging
 import org.mapdb.DB
 import org.mapdb.DBMaker
@@ -53,14 +55,7 @@ class HashStorage(
             // Ensure directory exists
             Files.createDirectories(dbPath.parent)
 
-            // Initialize MapDB with optimized settings
-            db =
-                DBMaker
-                    .fileDB(dbPath.toFile())
-                    .transactionEnable()
-                    .closeOnJvmShutdown()
-                    .fileMmapEnable()
-                    .make()
+            db = createDb(dbPath)
 
             // Create hash map for file content hashes
             hashMap =
@@ -177,3 +172,21 @@ class HashStorage(
         }
     }
 }
+
+private fun createDb(dbPath: Path): DB {
+    val dbMaker =
+        DBMaker
+            .fileDB(dbPath.toFile())
+            .transactionEnable()
+            .closeOnJvmShutdown()
+
+    if (shouldUseMemoryMapping(PlatformDetector.current)) {
+        dbMaker.fileMmapEnable()
+    } else {
+        logger.info { "Для Windows отключено memory-mapped хранилище MapDB для обхода проблем с WAL/truncate" }
+    }
+
+    return dbMaker.make()
+}
+
+internal fun shouldUseMemoryMapping(platform: PlatformType): Boolean = platform != PlatformType.WINDOWS

--- a/src/main/kotlin/io/github/alkoleft/mcp/infrastructure/storage/HashStorage.kt
+++ b/src/main/kotlin/io/github/alkoleft/mcp/infrastructure/storage/HashStorage.kt
@@ -173,14 +173,14 @@ class HashStorage(
     }
 }
 
-private fun createDb(dbPath: Path): DB {
-    val dbMaker =
+private fun createDb(dbPath: Path, platform: PlatformType = PlatformDetector.current): DB {
+    val dbMaker: DBMaker.Maker =
         DBMaker
             .fileDB(dbPath.toFile())
             .transactionEnable()
             .closeOnJvmShutdown()
 
-    if (shouldUseMemoryMapping(PlatformDetector.current)) {
+    if (canUseMemoryMapping(platform)) {
         dbMaker.fileMmapEnable()
     } else {
         logger.info { "Для Windows отключено memory-mapped хранилище MapDB для обхода проблем с WAL/truncate" }
@@ -189,4 +189,4 @@ private fun createDb(dbPath: Path): DB {
     return dbMaker.make()
 }
 
-internal fun shouldUseMemoryMapping(platform: PlatformType): Boolean = platform != PlatformType.WINDOWS
+internal fun canUseMemoryMapping(platform: PlatformType): Boolean = platform != PlatformType.WINDOWS

--- a/src/test/kotlin/io/github/alkoleft/mcp/infrastructure/storage/MapDbHashStorageTest.kt
+++ b/src/test/kotlin/io/github/alkoleft/mcp/infrastructure/storage/MapDbHashStorageTest.kt
@@ -1,0 +1,40 @@
+/*
+ * This file is part of METR.
+ *
+ * Copyright (C) 2025 Aleksey Koryakin <alkoleft@gmail.com> and contributors.
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * METR is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * METR is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with METR.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.alkoleft.mcp.infrastructure.storage
+
+import io.github.alkoleft.mcp.application.core.PlatformType
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+class HashStorageMapDbModeTest {
+    @Test
+    fun `should disable memory mapping on Windows`() {
+        assertFalse(shouldUseMemoryMapping(PlatformType.WINDOWS))
+    }
+
+    @Test
+    fun `should keep memory mapping on non-Windows platforms`() {
+        assertTrue(shouldUseMemoryMapping(PlatformType.LINUX))
+        assertTrue(shouldUseMemoryMapping(PlatformType.MACOS))
+    }
+}

--- a/src/test/kotlin/io/github/alkoleft/mcp/infrastructure/storage/MapDbHashStorageTest.kt
+++ b/src/test/kotlin/io/github/alkoleft/mcp/infrastructure/storage/MapDbHashStorageTest.kt
@@ -22,19 +22,46 @@
 package io.github.alkoleft.mcp.infrastructure.storage
 
 import io.github.alkoleft.mcp.application.core.PlatformType
-import org.junit.jupiter.api.Assertions.assertFalse
-import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Test
 
 class HashStorageMapDbModeTest {
     @Test
-    fun `should disable memory mapping on Windows`() {
-        assertFalse(shouldUseMemoryMapping(PlatformType.WINDOWS))
+    fun shouldDisableMemoryMappingOnWindows(): Unit {
+        // Arrange
+        val inputPlatform = PlatformType.WINDOWS
+        val expectedResult = false
+
+        // Act
+        val actualResult = canUseMemoryMapping(inputPlatform)
+
+        // Assert
+        assertEquals(expectedResult, actualResult)
     }
 
     @Test
-    fun `should keep memory mapping on non-Windows platforms`() {
-        assertTrue(shouldUseMemoryMapping(PlatformType.LINUX))
-        assertTrue(shouldUseMemoryMapping(PlatformType.MACOS))
+    fun shouldKeepMemoryMappingOnLinux(): Unit {
+        // Arrange
+        val inputPlatform = PlatformType.LINUX
+        val expectedResult = true
+
+        // Act
+        val actualResult = canUseMemoryMapping(inputPlatform)
+
+        // Assert
+        assertEquals(expectedResult, actualResult)
+    }
+
+    @Test
+    fun shouldKeepMemoryMappingOnMacOs(): Unit {
+        // Arrange
+        val inputPlatform = PlatformType.MACOS
+        val expectedResult = true
+
+        // Act
+        val actualResult = canUseMemoryMapping(inputPlatform)
+
+        // Assert
+        assertEquals(expectedResult, actualResult)
     }
 }

